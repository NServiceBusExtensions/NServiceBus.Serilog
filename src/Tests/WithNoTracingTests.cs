[UsesVerify]
public class WithNoTracingTests
{
    [Fact]
    public async Task Handler()
    {
        Exception? exception = null;
        var resetEvent = new ManualResetEvent(false);
        var configuration = ConfigBuilder.BuildDefaultConfig("WithNoTracingTests");
        configuration.DisableRetries();
        configuration.RegisterComponents(components => components.RegisterSingleton(resetEvent));

        var recoverability = configuration.Recoverability();
        recoverability.Failed(settings => settings
            .OnMessageSentToErrorQueue(message =>
            {
                exception = message.Exception;
                resetEvent.Set();
                return Task.CompletedTask;
            }));

        var endpoint = await Endpoint.Start(configuration);
        await endpoint.SendLocal(new StartHandler());
        if (!resetEvent.WaitOne(TimeSpan.FromSeconds(2)))
        {
            throw new("No Set received.");
        }

        await endpoint.Stop();
        await Verify(exception!.Message);
    }
}