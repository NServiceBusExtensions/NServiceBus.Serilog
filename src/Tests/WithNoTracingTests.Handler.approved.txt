Expected to find a `ILogger` in the pipeline context.
 * If this is a production exception, it is possible NServiceBus.Serilog has not been enabled using a call to `SerilogTracingExtensions.EnableSerilogTracing()`.
 * If this is a unit test exception, it is possible an `ILogger` has not been injected into the test context. eg `context.Extensions.Set(Log.Logger);`.